src/appengine/handlers/upload_testcase.py

```diff
@@ -183,7 +188,10 @@ def find_fuzz_target(engine, target_name, job_name):
 
   target = data_handler.get_fuzz_target(candidate_name)
   if not target:
-    raise helpers.EarlyExitException('Fuzz target does not exist.', 400)
+    if email and _is_trusted_uploader_allowed(email):
```

Comment:
Can you move this check out of this function? It's a side effect that doesn't fit well with this `find_fuzz_target` function. 

Instead, we could move this to line 355 right?